File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int nSize = needle.size();
        int hSize = haystack.size();
        if(nSize > hSize) return -1;
        if(nSize == hSize) {
            return (haystack == needle ? 0 : -1);
        }
        for(int i = 0; i <= hSize - nSize; i++) {
            string substr = haystack.substr(i, nSize);
            if(substr == needle) return i;
        }
        return -1;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        if(haystack.length() < needle.length()) {
            return -1;
        }
        boolean found = false;
        for(int i = 0; i < haystack.length(); i++){
            if(haystack.charAt(i) == needle.charAt(0)) {
                int l = i;
                found = true;
                for(int j = 0; j < needle.length(); j++) {
                    if(l >= haystack.length() || haystack.charAt(l) != needle.charAt(j)) {
                        found = false;
                        break;
                    }
                    l++;
                }
                if(found) {
                    return i;
                }
            }
        }
        return -1;
    }
}